src/features/account/AccountTEEndorsement.test.tsx

```diff
@@ -142,4 +178,206 @@ describe('AccountTEEndorsement', () => {
 
     expect(queryByText('mock.ens.name')).not.toBeInTheDocument();
   });
+
+  describe('when trigger onSign', () => {
+    it('calls `getSignedAssertion` when `signMessage` return signature', async () => {
+      buildEnsNameMock('mock.ens.name');
+      buildToastSpy();
+      const {
+        signMessageSpy,
+        getSignedAssertionSpy,
+        buildTechnicalExpertiseTrustSpy,
+      } = buildUseVerifiableCredentialMock(undefined);
+
+      signMessageSpy.mockResolvedValue('0xSignature');
+
+      await runOnSign();
+
+      expect(signMessageSpy).toHaveBeenCalled();
+      expect(getSignedAssertionSpy).toHaveBeenCalled();
+      expect(buildTechnicalExpertiseTrustSpy).toHaveBeenCalledWith(
+        VALID_ACCOUNT_2,
+        VALID_ACCOUNT_1,
+        [
+          {
+            scope: TrustworthinessScope.SoftwareDevelopment,
+            level: 0,
+          },
+          {
+            scope: TrustworthinessScope.SoftwareSecurity,
+            level: 0,
+          },
+        ],
+      );
+    });
+
+    it('does not calls `getSignedAssertion` when `signMessage` return null', async () => {
+      buildEnsNameMock('mock.ens.name');
+      buildToastSpy();
+      const {
+        signMessageSpy,
+        getSignedAssertionSpy,
+        buildTechnicalExpertiseTrustSpy,
+      } = buildUseVerifiableCredentialMock(undefined);
+
+      signMessageSpy.mockResolvedValue(null);
+
+      await runOnSign();
+
+      expect(signMessageSpy).toHaveBeenCalled();
+      expect(buildTechnicalExpertiseTrustSpy).toHaveBeenCalled();
+      expect(getSignedAssertionSpy).not.toHaveBeenCalled();
+    });
+
+    it('merge selected `ExpertiseTrust` and default `ExpertiseTrust`', async () => {
+      buildEnsNameMock('mock.ens.name');
+      buildToastSpy();
+      const {
+        signMessageSpy,
+        getSignedAssertionSpy,
+        buildTechnicalExpertiseTrustSpy,
+      } = buildUseVerifiableCredentialMock(undefined);
+
+      signMessageSpy.mockResolvedValue('0xSignature');
+
+      const { getByText } = render(
+        <AccountTEEndorsement
+          address={VALID_ACCOUNT_1}
+          connectedAddress={VALID_ACCOUNT_2}
+        />,
+      );
+
+      await act(async () =>
```

Comment:
Why there's a need for double nested `act`s?

Reply:
it is to wait for the state update

Reply:
i find that the double act has apply everywhere, let me create another fix PR to fix them all in once